msrest/pipeline/universal.py

```diff
@@ -156,6 +157,11 @@ def deserialize_from_text(cls, data, content_type=None):
             # Explain to mypy the correct type.
             data_as_str = cast(str, data)
 
+            # Remove Byte Order Mark if present in string
+            bom = codecs.BOM_UTF8.decode(encoding='utf-8')
+            if data_as_str.startswith(bom):
```

Comment:
A few thinkings here:
- I think in the usual workflow, we don't need it, because you patched HTTPClientResponse to utf-8-sig
- I'm hesitant on performance. `bom` can be a constant, `startswith` is unnecessary (`lstrip` will return the initial str if not startswith anyway)

I'm thinking maybe not adding this code, and just document it "if you give me str, you must have removed the BOM first". bytes and stream scenario will take care of the BOM.

Thoughts?

Reply:
Changed as requested. I believe we have 3 options:
1. (BEST) We keep this as is (not best performance) for now and then we can create a PR for requests library. I say this because when debugging I saw that the encoding was `utf-8` and [apparent encoding from chardet](https://github.com/requests/requests/blob/bedd9284c9646e50c10b3defdf519d4ba479e2c7/requests/models.py#L727) was `utf-8-sig`. So we can just specify to give preference apparent encoding.
2. We try to create a global boolean flag within msrest to remove BOM only if set to `true`.
3. Just documented it. This option would not help me. I was able to create a SDK directly from swagger file from my fork, but if i use the master i need to forcefully set `response.encoding = 'utf-8-sig'`.

Reply:
Yes, sounds good.

Reply:
The PR for requests has been created and, if they merge it there, I believe all the changes here will not be required at all. :)